component/service/src/main/java/org/exoplatform/social/rest/impl/space/SpaceRestResourcesV1.java

```diff
@@ -182,6 +183,15 @@ public Response getSpaces(@Context UriInfo uriInfo,
         space.setHasBindings(spaceBindingService.isBoundSpace(space.getId()));
         
         SpaceEntity spaceInfo = EntityBuilder.buildEntityFromSpace(space, authenticatedUser, uriInfo.getPath(), expand);
+        List<UserSpaceBinding> userSpaceBinding = spaceBindingService.findUserSpaceBindingsBySpace(space.getId(), authenticatedUser);
+        spaceInfo.setIsUserBound(false);
+        if (userSpaceBinding.size() != 0) {
+          for (int i = 0; i < userSpaceBinding.size(); i++) {
+            if (authenticatedUser.equals(userSpaceBinding.get(i).getUser()) && !userSpaceBinding.get(i).isMemberBefore()) {
+              spaceInfo.setIsUserBound(true);
+            }
```

Comment:
This is unneeded I think, you have just to check the size of userSpaceBinding, and if it is bigger than zero, you can call the function org.exoplatform.social.core.binding.impl.GroupSpaceBindingServiceImpl#isUserBoundAndMemberBefore to check if the user is bound before, otherwise, you enable the leave button

Reply:
ok